### PR TITLE
improve Dotty build

### PR DESCRIPTION
**Pull request by larsrh**
* disable dottydoc
* uniform Travis CI configuration

Currently doesn't build because of some weirdness in the `docs` subproject. I'd appreciate some guidance there.

### CODE REVIEW

**Comment by larsrh** [comment]
This is what I get locally:

```
sbt:cats> + update
[info] Setting Scala version to 0.27.0-RC1 on 13 projects.
[info] Excluded 15 projects, run ++ 0.27.0-RC1 -v for more details.
[info] Reapplying settings...
[info] set current project to cats (in build file:/home/lars/proj/typelevel/cats/)
[warn] There may be incompatibilities among your library dependencies; run 'evicted' to see detailed eviction warnings.
[error] Modules were resolved with conflicting cross-version suffixes in ProjectRef(uri("file:/home/lars/proj/typelevel/cats/"), "docs"):
[error]    org.typelevel:cats-laws _2.12, _0.27
[error]    org.typelevel:cats-free _2.12, _0.27
[error]    org.typelevel:cats-kernel-laws _2.12, _0.27
[error]    org.typelevel:cats-kernel _2.12, _0.27
[error]    org.typelevel:cats-core _2.12, _0.27
[error] stack trace is suppressed; run last docs / update for the full output
[error] (docs / update) Conflicting cross-version suffixes in: org.typelevel:cats-laws, org.typelevel:cats-free, org.typelevel:cats-kernel-laws, org.typelevel:cats-kernel, org.typelevel:cats-core
[error] Total time: 1 s, completed Oct 31, 2020, 9:03:02 PM
```

File: build.sbt [hunk 1]
```diff
@@ -139,6 +146,7 @@ lazy val commonJvmSettings = Seq(
     val flag = if ((isTravisBuild in Global).value) "-oCI" else "-oDF"
     Tests.Argument(TestFrameworks.ScalaTest, flag)
   },
+  crossScalaVersions += "0.27.0-RC1",
```
Comment (LukaJCB) [inline build.sbt:149]: Maybe we should rename the `dotty_version` variable in the travis config to `scala_version_dotty`? WDYT?

Comment (larsrh) [reply build.sbt:149]: Yeah, sounds good. Or we shanghai @djspiewak into finishing #3570 😉

**Comment by larsrh** [comment]
Ready to go now :rocket:

File: build.sbt [hunk 2]
```diff
@@ -441,8 +453,7 @@ lazy val catsJVM = project
              alleycatsCore.jvm,
              alleycatsLaws.jvm,
              alleycatsTests.jvm,
-             jvm,
-             docs
```
Comment (LukaJCB) [inline build.sbt:445]: If remove docs from here, is it still going to be run in CI?

Comment (larsrh) [reply build.sbt:445]: Yes, it's a separate task in the Travis CI config.

**Comment by larsrh** [comment]
#3570

**Pull request closed without merging**